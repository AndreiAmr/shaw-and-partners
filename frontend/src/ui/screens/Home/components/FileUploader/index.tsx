import {
  Box,
  Button,
  Flex,
  Image,
  Input,
  Text,
  useDisclosure,
} from "@chakra-ui/react";
import ImageBackground from "../../../../assets/no-data.svg";
import { useContext, useRef } from "react";
import { useFileUploader } from "./hooks/useFileUploader";
import AlertDialog from "./components/AlertDialog";
import { UsersContext } from "../../../../../infra/contexts/UsersContexts";

const FileUploader = () => {
  const { isLoading, setIsLoading } = useContext(UsersContext);
  const { handlers, items } = useFileUploader();
  const inputRef = useRef<HTMLInputElement>(null);
  const { isOpen, onClose, onOpen } = useDisclosure();

  return (
    <div data-testid="meucu">
      <Flex flexDirection="column" data-testid="file-uploader-container">
        <Box
          data-testid="file-uploader-container"
          border="3px dashed #EFEFEF"
          padding="24px"
          borderRadius="10px"
          position={"relative"}
          display="flex"
          alignItems="center"
          justifyContent="center"
          flexDirection="column"
        >
          <Image src={ImageBackground} height="25vh" data-testid="my-image" />
          <Text
            fontSize={["1.5rem"]}
            fontWeight="semibold"
            color="white"
            textAlign="center"
          >
            {isLoading
              ? "Loading your file"
              : "Drop here your CSV file or click the button bellow"}
          </Text>
          <Input
            ref={inputRef}
            type="file"
            accept="text/csv"
            position={"absolute"}
            bottom={0}
            top={0}
            left={0}
            right={0}
            opacity={0}
            height="100%"
            cursor="default"
            onChange={(e) => {
              e.preventDefault();

              handlers.handleSelectFile(e.target.files![0]);

              onOpen();
            }}
            onDrop={(e) => {
              e.preventDefault();
              handlers.handleSelectFile(e.dataTransfer.files[0]);

              onOpen();
            }}
          />
        </Box>

        <Button
          marginTop="3vh"
          width=""
          onClick={() => {
            inputRef.current?.click();
          }}
        >
          Choose file
        </Button>
        <AlertDialog
          isOpen={isOpen}
          onClose={onClose}
          filename={items.currentFile?.name || ""}
          onConfirm={() => {
            setIsLoading(true);
            handlers.handleOnConfirmDialog(onClose);
          }}
        />
      </Flex>
    </div>
  );
};

export default FileUploader;
