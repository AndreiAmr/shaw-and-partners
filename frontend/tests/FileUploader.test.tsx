import { render, screen } from "@testing-library/react";
import FileUploader from "../src/ui/screens/Home/components/FileUploader";

import "@testing-library/jest-dom";
import PackagesProvider from "../src/infra/packages";

jest.mock("@chakra-ui/react", () => {
  return {
    extendTheme: () => {},
    ChakraProvider: () => {},
    useDisclosure: () => ({
      isOpen: true,
      onClose: () => {},
      onOpen: () => {},
    }),
  };
});

test("File Uploader", async () => {
  const { container } = render(<FileUploader />, {
    wrapper: PackagesProvider,
  });

  screen.debug(container);

  expect(container).toMatchSnapshot();
});
