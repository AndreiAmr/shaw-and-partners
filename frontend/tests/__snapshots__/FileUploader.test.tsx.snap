// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Uploader 1`] = `<div />`;
